src/hooks/UseRepositoryPermissions.ts

```diff
@@ -0,0 +1,96 @@
+import {useQuery} from '@tanstack/react-query';
+import {useState} from 'react';
+import {SearchState} from 'src/components/toolbar/SearchTypes';
+import {
+  fetchTeamRepoPermissions,
+  fetchUserRepoPermissions,
+  RepoMember,
+} from 'src/resources/RepositoryResource';
+import ColumnNames from 'src/routes/RepositoryDetails/Settings/ColumnNames';
+
+enum MemberType {
+  user = 'user',
+  robot = 'robot',
+  team = 'team',
+}
+
+export function useRepositoryPermissions(org: string, repo: string) {
+  const [page, setPage] = useState(1);
+  const [perPage, setPerPage] = useState(10);
+  const [search, setSearch] = useState<SearchState>({
+    query: '',
+    field: ColumnNames.account,
+  });
+
+  const {
+    data: userRoles,
+    isError: errorLoadingUserRoles,
+    isLoading: loadingUserRoles,
+  } = useQuery(
+    ['userrepopermissions', org, repo],
+    () => fetchUserRepoPermissions(org, repo),
+    {
+      placeholderData: {},
```

Comment:
Can we put an empty array here?

Reply:
The return type is key-value pairs instead of an array.